src/components/modal/modal-wrapper.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { ReactNode, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import CloseIcon from "@danskernesdigitalebibliotek/dpl-design-system/build/icons/collection/CloseLarge.svg";
+import { closeModal, openModal } from "../../core/modal.slice";
+
+type ModalWrapperProps = {
+  children: ReactNode;
+  modalId: string;
+  closeModalAriaLabelText: string;
+  screenReaderModalDescriptionText: string;
+};
+
+interface ModalIdsProps {
+  modal: {
+    modalIds: string[];
+  };
+}
+
+function ModalWrapper({
+  modalId,
+  children,
+  closeModalAriaLabelText,
+  screenReaderModalDescriptionText
+}: ModalWrapperProps) {
+  const dispatch = useDispatch();
+  const { modalIds } = useSelector((s: ModalIdsProps) => s.modal);
+
+  useEffect(() => {
+    const searchParams = new URLSearchParams(window.location.search);
```

Comment:
I think we should do this with a helper funktion that is on its way to your review.
Although I can see now that I missed using the url parameter available as an parameter for the function:
https://github.com/danskernesdigitalebibliotek/dpl-react/pull/33/files#diff-b388ec997dbc272cfdf0283ac04a0762401ff68fab0d3c9eba31a132b581e18bR11